File: src/std/js/AtomicBool.js


fan.std.AtomicBool = fan.sys.Obj.$extend(fan.sys.Obj);
fan.std.AtomicBool.prototype.$ctor = function() {}
fan.std.AtomicBool.prototype.$typeof = function() { return fan.std.AtomicBool.$type; }

fan.std.AtomicBool.make = function(v) {
	self = new fan.std.AtomicBool();
	if (v === undefined) v = false;
	self.m_val = v;
	return self;
}

fan.std.AtomicBool.prototype.get = function() {
	return this.m_val;
}

fan.std.AtomicBool.prototype.val = function() {
	return this.m_val;
}

fan.std.AtomicBool.prototype.val$ = function(v) {
	this.m_val = v;
}

fan.std.AtomicBool.prototype.set$ = function(v) {
	this.m_val = v;
}

fan.std.AtomicBool.prototype.getAndSet = function(v) {
	o = this.m_val;
	this.m_val = v;
	return o;
}

fan.std.AtomicBool.prototype.compareAndSet = function(expect, update) {
	if (this.m_val == expect) {
		this.m_val = update;
		return true;
	}
	return false;
}

File: library/std/java/src/fan/std/AtomicBool.java
//
// Copyright (c) 2010, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   12 Feb 10 Brian Frank  Creation
//
package fan.std;

import java.util.concurrent.atomic.AtomicBoolean;

import fan.sys.FanObj;
import fanx.main.Sys;
import fanx.main.Type;

public final class AtomicBool extends FanObj {
	private static Type type = null;

	public Type typeof() {
		if (type == null) {
			type = Sys.findType("std::AtomicBool");
		}
		return type;
	}
	
	public AtomicBool(boolean v) {
		this.val = new AtomicBoolean(v);
	}
	
	public static AtomicBool make() {
		return new AtomicBool(false);
	}

	public static AtomicBool make(boolean v) {
		return new AtomicBool(v);
	}
	
	public final boolean get() {
		return val.get();
	}

	public final boolean val() {
		return val.get();
	}

	public final void val(boolean v) {
		val.set(v);
	}
	
	public final void get(boolean v) {
		val.set(v);
	}

	public final boolean getAndSet(boolean v) {
		return val.getAndSet(v);
	}

	public final boolean compareAndSet(boolean expect, boolean update) {
		return val.compareAndSet(expect, update);
	}
	
	public String toStr() {
		return val.toString();
	}
	
	private final AtomicBoolean val;
}